projects/components/src/utils/test/angular-widget-driver.ts

```diff
@@ -24,90 +24,81 @@ export class AngularWidgetDriver<T> implements WidgetDriver {
+    async click(cssSelector?: string): Promise<void> {
         this.getDebugElement(cssSelector).nativeElement.click();
         this.fixture.detectChanges();
-        return Promise.resolve();
     }
 
-    classes(cssSelector?: string): Promise<string[]> {
-        return Promise.resolve(Object.keys(this.getDebugElement(cssSelector).nativeElement.classes));
+    async classes(cssSelector?: string): Promise<string[]> {
+        return Object.keys(this.getDebugElement(cssSelector).nativeElement.classes);
     }
 
-    check(cssSelector?: string): Promise<void> {
+    async check(cssSelector?: string): Promise<void> {
         const checkboxElement = this.getDebugElement(cssSelector).nativeElement;
         checkboxElement.checked = !checkboxElement.checked;
         checkboxElement.dispatchEvent(new Event('change'));
         this.fixture.detectChanges();
-        return Promise.resolve();
     }
 
-    getChecked(cssSelector?: string): Promise<boolean> {
-        return Promise.resolve(this.getDebugElement(cssSelector).nativeElement.checked);
+    async getChecked(cssSelector?: string): Promise<boolean> {
+        return this.getDebugElement(cssSelector).nativeElement.checked;
     }
 
-    select(option: string, cssSelector?: string): Promise<void> {
+    async select(option: string, cssSelector?: string): Promise<void> {
         const selectElement = this.getDebugElement(cssSelector).nativeElement;
         selectElement.value = option;
         selectElement.dispatchEvent(new Event('change'));
         this.fixture.detectChanges();
-        return Promise.resolve();
     }
 
-    sendKeyboardKey(key: string, cssSelector?: string): Promise<void> {
+    async sendKeyboardKey(key: string, cssSelector?: string): Promise<void> {
         const nativeElement: HTMLBaseElement = this.getDebugElement(cssSelector).nativeElement;
         nativeElement.dispatchEvent(new KeyboardEvent('keydown', { key, bubbles: true }));
         nativeElement.dispatchEvent(new KeyboardEvent('keyup', { key, bubbles: true }));
         this.fixture.detectChanges();
-        return Promise.resolve();
     }
 
-    setInputValue(value: string, cssSelector?: string): Promise<void> {
+    async setInputValue(value: string, cssSelector?: string): Promise<void> {
         const nativeElement: HTMLInputElement | HTMLTextAreaElement = this.getDebugElement(cssSelector).nativeElement;
         nativeElement.value = String(value);
         nativeElement.dispatchEvent(new Event('input'));
         this.fixture.detectChanges();
-        return Promise.resolve();
     }
 
-    clearInputValue(cssSelector?: string): Promise<void> {
+    async clearInputValue(cssSelector?: string): Promise<void> {
         return this.setInputValue('', cssSelector);
     }
 
-    getText(cssSelector?: string): Promise<string> {
-        return this.getTexts(undefined, cssSelector).then(texts => texts[0]);
+    async getText(cssSelector?: string): Promise<string> {
```

Comment:
This should be implemented using `getDebugElement()` for performance reasons (so it stops searching after it finds one)